grammar.js

```diff
@@ -62,6 +62,10 @@ module.exports = grammar({
     [$._unannotated_type, $.scoped_type_identifier],
     [$._unannotated_type, $.generic_type],
     [$.generic_type, $.primary_expression],
+    // Switch constructs can be both expressions and statements
+    [$.expression, $.statement],
```

Comment:
It would be good to see what a canonical java parser (such as [`JavaParser`](https://javaparser.org/)) does when it sees this type of conflict (a switch expression followed by a token like `+` at the beginning of a statement). For example:

```java
static void foo(int i, int j) {
    // switch expressions inside of a binary expression
    switch (i) { default:  yield 1; } + switch (j) { default: yield 2; };
    
    // traditional switch statement, followed by another noop statement starting with an operator
    switch (i) { case 1: return; }
    + i;
}
```

Are both of these allowed? If not, maybe we could avoid this conflict using a precedence of some kind.

Reply:
Java doesn't actually allow arbitrary expressions to be ExpressionStatements, so for example `+ i;` is not valid. (ExpressionStatement spec: https://docs.oracle.com/javase/specs/jls/se16/html/jls-14.html#jls-14.8). If the reason switch statements aren't working is because of that specific type of example, a reasonable alternative could be to revise our expression_statement production?

Reply:
It would probably be difficult to restructure `expression_statement`, but I think this means that, in the event of a local ambiguity like this, we can safely prefer to parse a `switch_expression` as an *expression*, as opposed to parsing it as a statement. So we can put a greater precedence in the `_expression` rule.

Maybe give something like this a try:

```js
_expression: $ => choice(
  // ...
  prec(1, $.switch_expression)
)
```